html2/tokenizer.cpp

```diff
@@ -214,6 +236,507 @@ void Tokenizer::run() {
+                        state_ = State::ScriptDataDoubleEscapedLessThanSign;
+                        emit(CharacterToken{*c});
+                        continue;
+                    case '\0':
+                        // This is an unexpected-null-character parse error.
+                        state_ = State::ScriptDataDoubleEscaped;
+                        // TODO(mkiael): Emit replacement character for null
+                        // emit(CharacterToken("\xFF\xFD"));
+                        continue;
+                    default:
+                        state_ = State::ScriptDataDoubleEscaped;
+                        emit(CharacterToken{*c});
+                        continue;
+                }
+            }
+
+            case State::ScriptDataDoubleEscapedDashDash: {
+                auto c = consume_next_input_character();
+                if (!c) {
+                    // This is an eof-in-script-html-comment-like-text parse error.
+                    emit(EndOfFileToken{});
+                    return;
+                }
+
+                switch (*c) {
+                    case '-':
+                        emit(CharacterToken{*c});
+                        continue;
+                    case '<':
+                        state_ = State::ScriptDataDoubleEscapedLessThanSign;
+                        emit(CharacterToken{*c});
+                        continue;
+                    case '>':
+                        state_ = State::ScriptData;
+                        emit(CharacterToken{*c});
+                        continue;
+                    case '\0':
+                        // This is an unexpected-null-character parse error.
+                        state_ = State::ScriptDataDoubleEscaped;
+                        // TODO(mkiael): Emit replacement character for null
+                        // emit(CharacterToken("\xFF\xFD"));
+                        continue;
+                    default:
+                        state_ = State::ScriptDataDoubleEscaped;
+                        emit(CharacterToken{*c});
+                        continue;
+                }
+            }
+
+            case State::ScriptDataDoubleEscapedLessThanSign: {
+                auto c = consume_next_input_character();
+                if (c == '/') {
+                    temporary_buffer_ = "";
+                    state_ = State::ScriptDataDoubleEscapeEnd;
+                    emit(CharacterToken{*c});
+                    continue;
+                }
+
+                reconsume_in(State::ScriptDataDoubleEscaped);
+                continue;
+            }
+
+            case State::ScriptDataDoubleEscapeEnd: {
+                auto c = consume_next_input_character();
+                if (!c) {
+                    reconsume_in(State::ScriptDataDoubleEscaped);
+                    continue;
+                }
+
+                if (is_ascii_upper_alpha(*c)) {
+                    auto lower_case_c = to_lower(*c);
```

Comment:
I think it's more readable if you inline the `to_lower` into the `append` call. No need for a temporary variable when it's only used once.

Reply:
Did you intend to change this? :P

Reply:
I was! Fixed now.